src/Collector/AbstractCollector.php

```diff
@@ -15,19 +17,31 @@ abstract class AbstractCollector implements CollectorInterface, \Serializable
      */
     protected $data;
 
+    public function __serialize()
+    {
+        return serialize($this->data);
+    }
```

Comment:
@snapshotpl Shouldn't this return an array instead of a serialized string? From https://www.php.net/manual/en/language.oop5.magic.php:

> `public __serialize(): array`
> `public __unserialize(array $data): void`
> 
> serialize() checks if the class has a function with the magic name __serialize(). If so, that function is executed prior to any serialization. It must construct and return an associative array of key/value pairs that represent the serialized form of the object. If no array is returned a TypeError will be thrown.

Reply:
@driehle 
Good catch! 👍 
Please create an issue report or a pull request for this problem. Thanks in advance!